packages/diagrams/frontend/sirius-components-diagrams/src/renderer/node/FreeFormNode.tsx

```diff
@@ -93,10 +124,17 @@ const computeBorderRotation = (data: FreeFormNodeData): string | undefined => {
 export const FreeFormNode: NodeComponentsMap['freeFormNode'] = memo(
   ({ data, id, selected, dragging }: NodeProps<Node<FreeFormNodeData>>) => {
     const { httpOrigin } = useContext<ServerContextValue>(ServerContext);
+    const { classes } = useStyles();
 
     const theme = useTheme();
     const { onDrop, onDragOver } = useDrop();
 
+    const { invokeAction, actions } = useActions({
```

Comment:
Anytime a node is rendered you will perform a request to ask for its actions? That will be VERY expensive, asking for the actions should be done in the action container, this way you will only display the action container if the node is hovered and the request will only be performed if the node is hovered.

It will also remove the need for every custom node in every downstream application to manage the retrieval of the actions

Reply:
done